homeassistant/components/here_weather/config_flow.py

```diff
@@ -0,0 +1,244 @@
+    """Get the here_weather location_name schema."""
+    schema = get_base_schema(hass)
+    return schema.extend({vol.Required(CONF_LOCATION_NAME): str})
+
+
+async def async_validate_coordinate_input(
+    hass: HomeAssistant, user_input: dict
+) -> None:
+    """Validate the user_input containing coordinates."""
+    await async_validate_name(hass, user_input)
+    here_client = herepy.DestinationWeatherApi(user_input[CONF_API_KEY])
+    await hass.async_add_executor_job(
+        here_client.weather_for_coordinates,
+        user_input[CONF_LATITUDE],
+        user_input[CONF_LONGITUDE],
+        herepy.WeatherProductType[user_input[CONF_MODE]],
+    )
+
+
+async def async_validate_zip_code_input(hass: HomeAssistant, user_input: dict) -> None:
+    """Validate the user_input containing a zip_code."""
+    await async_validate_name(hass, user_input)
+    here_client = herepy.DestinationWeatherApi(user_input[CONF_API_KEY])
+    await hass.async_add_executor_job(
+        here_client.weather_for_zip_code,
+        user_input[CONF_ZIP_CODE],
+        herepy.WeatherProductType[user_input[CONF_MODE]],
+    )
+
+
+async def async_validate_location_name_input(
+    hass: HomeAssistant, user_input: dict
+) -> None:
+    """Validate the user_input containing a location_name."""
+    await async_validate_name(hass, user_input)
+    here_client = herepy.DestinationWeatherApi(user_input[CONF_API_KEY])
+    await hass.async_add_executor_job(
+        here_client.weather_for_location_name,
+        user_input[CONF_LOCATION_NAME],
+        herepy.WeatherProductType[user_input[CONF_MODE]],
+    )
+
+
+async def async_validate_name(hass: HomeAssistant, user_input: dict) -> None:
+    """Validate the user input."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.data[CONF_NAME] == user_input[CONF_NAME]:
+            raise AlreadyConfigured
+
+
+class HereWeatherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for here_weather."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry):
+        """Get the options flow for this handler."""
+        return HereWeatherOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            if user_input[CONF_OPTION] == CONF_OPTION_COORDINATES:
+                return await self.async_step_coordinates()
+            if user_input[CONF_OPTION] == CONF_OPTION_ZIP_CODE:
+                return await self.async_step_zip_code()
+            if user_input[CONF_OPTION] == CONF_OPTION_LOCATION_NAME:
+                return await self.async_step_location_name()
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Required(CONF_OPTION): vol.In(CONF_OPTIONS)}),
```

Comment:
Can we pick one of the three location options instead? Having three options for the same thing makes the integration complicated.